libs/application/ui-fields/src/lib/CompanySearchFormField/CompanySearchFormField.tsx

```diff
@@ -0,0 +1,135 @@
+import { gql, useLazyQuery } from '@apollo/client'
+import {
+  CompanySearchField,
+  FieldBaseProps,
+  formatText,
+} from '@island.is/application/core'
+import { Box } from '@island.is/island-ui/core'
+import { useLocale } from '@island.is/localization'
+import debounce from 'lodash/debounce'
+import React, { FC, useMemo, useState } from 'react'
+import { CompanySearchItem } from './CompanySearchItem'
+import kennitala from 'kennitala'
+import { CompanySearchController } from '@island.is/shared/form-fields'
+import { mockCompanies } from './__mock-data__/companies'
+
+export const COMPANY_REGISTRY_COMPANIES = gql`
+  query SearchCompanies($input: RskCompanyInfoSearchInput!) {
+    companyRegistryCompanies(input: $input) {
+      data {
+        name
+        nationalId
+      }
+    }
+  }
```

Comment:
Would be great to have this logic in the shared form-field so it's easier to reuse.

The only thing the application ui-field should do is make it fit with the application system context.

Reply:
Moved all logic to shared form field